lib/wallet/signers/WalletConnect.js

```diff
@@ -1,7 +1,57 @@
+const {default: algosdk} = require('algosdk');
+const {formatJsonRpcRequest} = require('@json-rpc-tools/utils');
+const groupBy = require('../../utils/groupBy');
+
 /**
+ * WalletConnectSigner
  *
+ * @todo move to SDK
+ * @param {Array<Order>} orders A list of Compiled Orders
+ * @return {Promise<*>}
  */
-function signer() {
+async function signer(orders) {
+  const orderTxns = orders.map((execObj) => execObj.contract.txns);
+
+  orderTxns.forEach((outerTxns) => algosdk.assignGroupID(
+      outerTxns.map((txn) => txn.unsignedTxn),
+  ));
+
+
+  const outerTxns = [];
+  let txNum = 0; // txNum has to be assigned outside nested loop in order to preserve total order
+  orderTxns.forEach((txns, groupNum) => {
+    txns.forEach((txn) => {
+      outerTxns.push({...txn, groupNum: groupNum, txNum: txNum});
+      txNum++;
+    });
+  });
+
+  const encodedTxns = outerTxns.map((txn) => {
+    const encodedTxn = Buffer.from(algosdk.encodeUnsignedTransaction(txn.unsignedTxn)).toString('base64');
+    if (algosdk.encodeAddress(txn.unsignedTxn.from.publicKey) !== this.accounts[0]) return {txn: encodedTxn, signers: []}; // naive solution assuming target address is first item in accountds array
```

Comment:
I don't really understand this but I am guessing you know what you're doing and this works in all cases.